server/rest/src/main/java/org/infinispan/rest/resources/BackupManagerResource.java

```diff
@@ -161,18 +160,19 @@ static CompletionStage<RestResponse> handleRestore(InvocationHelper invocationHe
       Json resourcesJson = Json.object();
       MediaType contentType = request.contentType();
       boolean uploadedBackup = contentType.match(MediaType.MULTIPART_FORM_DATA);
+      final HttpPostMultipartRequestDecoder decoder;
       try {
          if (uploadedBackup) {
             FullHttpRequest nettyRequest = ((NettyRestRequest) request).getFullHttpRequest();
             DefaultHttpDataFactory factory = new DefaultHttpDataFactory(true);
-            InterfaceHttpPostRequestDecoder decoder = new HttpPostMultipartRequestDecoder(factory, nettyRequest);
+            decoder = new HttpPostMultipartRequestDecoder(factory, nettyRequest);
             DiskFileUpload backup = (DiskFileUpload) decoder.getBodyHttpData("backup");
             path = backup.getFile().toPath();
-
             DiskAttribute resources = (DiskAttribute) decoder.getBodyHttpData("resources");
             if (resources != null)
                resourcesJson = Json.read(resources.getString());
          } else if (contentType.match(MediaType.APPLICATION_JSON)) {
+            decoder = null;
```

Comment:
maybe we should add this set also to the other `else if` branch

Reply:
Not necessary, because the other branch returns early, skipping the actual decoder.